src/main/java/com/openlattice/chronicle/services/surveys/SurveysService.java

```diff
@@ -436,4 +444,547 @@ public void submitAppUsageSurvey(
+        return ImmutableMap.of(
+                edmCacheManager.getPropertyTypeId( START_DATE_TIME_FQN ), ImmutableSet.of( dateTime ),
+                edmCacheManager.getPropertyTypeId( STRING_ID_FQN ), ImmutableSet.of( studyId ),
+                edmCacheManager.getPropertyTypeId( PERSON_ID_FQN ), ImmutableSet.of( participantId )
+        );
+    }
+
+    private Map<UUID, Set<Object>> getTimeRangeEntity( Map<FullQualifiedName, Set<Object>> data ) {
+        Set<Object> startDatetime = data.getOrDefault( START_DATE_TIME_FQN, ImmutableSet.of() );
+        Set<Object> endDatetime = data.getOrDefault( END_DATE_TIME_FQN, ImmutableSet.of() );
+
+        return ImmutableMap.of(
+                edmCacheManager.getPropertyTypeId( START_DATE_TIME_FQN ), startDatetime,
+                edmCacheManager.getPropertyTypeId( END_DATE_TIME_FQN ), endDatetime
+        );
+    }
+
+    private Map<UUID, Set<Object>> getSubmissionEntity( UUID studyId, String participantId, OffsetDateTime dateTime ) {
+        Map<UUID, Set<Object>> entity = Maps.newHashMap();
+
+        entity.put( edmCacheManager.getPropertyTypeId( DATE_TIME_FQN ), ImmutableSet.of( dateTime ) );
+        entity.put( edmCacheManager.getPropertyTypeId( STRING_ID_FQN ), ImmutableSet.of( studyId ) );
+        entity.put( edmCacheManager.getPropertyTypeId( OL_ID_FQN ), ImmutableSet.of( participantId ) );
+
+        return entity;
+    }
+
+    private Map<UUID, Set<Object>> getAddressesEntity( OffsetDateTime dateTime ) {
+        return ImmutableMap.of(
+                edmCacheManager.getPropertyTypeId( COMPLETED_DATE_TIME_FQN ),
+                ImmutableSet.of( dateTime )
+        );
+    }
+
+    private Map<UUID, Set<Object>> getQuestionPartOfEntity( Map<FullQualifiedName, Set<Object>> data ) {
+        return ImmutableMap.of(
+                edmCacheManager.getPropertyTypeId( OL_ID_FQN ), data.get( OL_ID_FQN )
+        );
+    }
+
+    private Map<UUID, Set<Object>> getSurveyPartOfEntityData( UUID studyId ) {
+        return ImmutableMap.of(
+                edmCacheManager.getPropertyTypeId( OL_ID_FQN ), ImmutableSet.of( studyId )
+        );
+    }
+
+    // associate entityKeys with EKIDS
+    private Map<EntityKey, UUID> createEntityKeyIdMap(
+            Set<EntityKey> entityKeys,
+            Set<Triple<EntityKey, EntityKey, EntityKey>> edgesByEntityKey,
+            ChronicleCoreAppConfig coreAppConfig,
+            DataIntegrationApi integrationApi,
+            UUID studyId,
+            UUID studyEKID,
+            UUID participantEKID,
+            String participantId
+    ) {
+        Map<EntityKey, UUID> entityKeyIdMap = Maps.newHashMap();
+
+        Set<EntityKey> orderedEntityKeys = Sets.newHashSet( entityKeys );
+
+        // entity set ids
+        UUID studyEntitySetId = coreAppConfig.getStudiesEntitySetId();
+        UUID participantEntitySetId = coreAppConfig.getParticipantEntitySetId();
+
+        edgesByEntityKey.forEach( triple -> {
```

Comment:
what is the order of this triple? it might be helpful to indicate that expected order somewhere